### PR TITLE
Add Server Save configuration to the config.lua file

### PR DESCRIPTION
**Pull request by marcomoa**
Add configuration to enable/disable server save notify

Co-Authored-By: Renato Foot Guimarães Costallat <costallat@hotmail.com>

### CODE REVIEW

File: config.lua.dist [hunk 1]
```diff
@@ -76,6 +76,12 @@ classicEquipmentSlots = false
 classicAttackSpeed = false
 showScriptsLogInConsole = true
 
+-- Server Save
+notifyServerSave = true
```
Comment (DSpeichert) [inline config.lua.dist]: `notifyAtServerSave` is probably more correct

File: config.lua.dist [hunk 2]
```diff
@@ -76,6 +76,12 @@ classicEquipmentSlots = false
...
+cleanMapAtServerSave = false
+closeAtServerSave = false
+shutdownAtServerSave = true -- If you set close and shutdown to 'true' it will close and then shutdown the server.
```
Comment (DSpeichert) [inline config.lua.dist]: This comment doesn't really say anything more than the variable names already say

File: data/globalevents/scripts/serversave.lua [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
-	return not shutdownAtServerSave
-end
+	if configManager.getBoolean(configKeys.NOTIFY_SERVER_SAVE) then
+		Game.broadcastMessage("Server is saving game in 5 minutes. Please logout.", MESSAGE_STATUS_WARNING)
+	end
+
+	addEvent(ServerSaveWarning, 60000, 300000)
+
+	return not configManager.getBoolean(configKeys.SHUTDOWN_AT_SERVER_SAVE)
+end
```
Comment (EPuncker) [inline data/globalevents/scripts/serversave.lua]: please don't remove new line at the end of file

**Comment by dudantas** [comment]
This PR is from OTServBR - Global repository 

My advice is not to merge it, it created a bug in our repository, it could only put 5 items in the backpack and it bugged the slots.
My advice is that you do not copy commits from our repository even before you merge, because our repository is new and we are liable to make mistakes.

File: src/configmanager.cpp [hunk 4]
```diff
@@ -177,6 +177,12 @@ bool ConfigManager::load()
 	integer[MAX_MARKET_OFFERS_AT_A_TIME_PER_PLAYER] = getGlobalNumber(L, "maxMarketOffersAtATimePerPlayer", 100);
 	integer[MAX_PACKETS_PER_SECOND] = getGlobalNumber(L, "maxPacketsPerSecond", 25);
 
+	boolean[SERVER_SAVE_NOTIFY_MESSAGE] = getGlobalBoolean(L, "serverSaveNotifyMessage", true);
+	integer[SERVER_SAVE_NOTIFY_DURATION] = getGlobalNumber(L, "serverSaveNotifyDuration", 5);
```
Comment (dudantas) [inline src/configmanager.cpp]: By our tests, this was the function that caused the bug I mentioned in the conversation.

**Comment by Znote** [comment]
@dudantas Can you pull this PR and see if it works fine?
https://github.com/marcomoa/forgottenserver/tree/serversave

OtservBR has lots of experimental code, to test PR's one by one in this repository is a nice way to isolate and fix those bugs before they get merged.

**Comment by dudantas** [comment]
> @dudantas Can you pull this PR and see if it works fine?
> https://github.com/marcomoa/forgottenserver/tree/serversave
> 
> OtservBR has lots of experimental code, to test PR's one by one in this repository is a nice way to isolate and fix those bugs before they get merged.

He took an experimental code and brought it here, that hadn't been merged yet.. As I mentioned, it is a mistake to do this ... I think he should test the code to see if everything is all right here. I just felt responsible for warning that there was a bug there, it may not happen the same here because of the difference in the source code.

**Comment by slavidodo** [comment]
I'm against using a config just to access them from a script.
You can use have a global.config.lua besides globa.lua or move the whole saving code to c++.

**Comment by dudantas** [comment]
You want to talk about a bugs of the repository (and we don't deny that it does) and want to bring one code in development, of there to here? Really do not understand.

**Comment by marcomoa** [comment]
> You want to talk about a bugs of the repository (and we don't deny that it does) and want to bring one code in development, of there to here? Really do not understand...

Not all fruits on a tree are rotten.
Avoid parallel conversations on pull request please.

**Comment by EPuncker** [comment]
I tested these changes (changed 60000 into 6000 for faster testing, and
<globalevent name="Server Save" interval="6000" script="serversave.lua" />
and:
```
serverSaveNotifyDuration = 2
serverSaveCleanMap = true
serverSaveClose = false
serverSaveShutdown = false
```

and guess what? working fine, no slot bugs whatsoever:
iH1VxGX
9K3nBj2

although I kinda agree with @slavidodo about moving save function to source (again?!) or we could get a new (and less uglier) way to add new settings into config.lua without the need to source edit (or maybe even a "global"config.lua that would work along with the config.lua but would use only global variables as settings)

**Pull request merged**